compiler/xc32/compiler.yml

```diff
@@ -19,18 +19,19 @@
 
 compiler.path.cc: "xc32-gcc"
 compiler.path.as: "xc32-gcc"
+compiler.path.cpp: "xc32-g++"
 compiler.path.archive: "xc32-ar"
 compiler.path.objdump: "xc32-objdump"
 compiler.path.objsize: "xc32-size"
 compiler.path.objcopy: "xc32-objcopy"
 
-compiler.flags.base: [-std=gnu11, -msmart-io=0, -fno-common]
+compiler.flags.base: [-std=gnu11, -fno-common, -mnewlib-libc, -ffunction-sections, -fdata-sections]
 compiler.flags.default: [compiler.flags.base, -O2, -g3]
-compiler.flags.optimized: [compiler.flags.base, -Os, -g3]
-compiler.flags.debug: [compiler.flags.base, -g3]
+compiler.flags.optimized: [compiler.flags.base, -O2, -g3]
```

Comment:
Wouldn't it be better to use `-Os` for optimized (for size!) and `-O2` (or 3) for `compiler.flags.speed` like it's done for ARM?

Reply:
Free compiler from microchip is limited when -Os is selected:
`cc1.exe: note: Disable the option or visit http://www.microchip.com/MPLABXCcompilers to purchase a new MPLAB XC compiler license.
repos/apache-mynewt-core/hw/hal/src/sbrk_sim.c:1:0: warning: Compiler option (Optimize for size) ignored because this feature requires the MPLAB XC32 PRO compiler [enabled by default]
`

To avail this warning -O2 is selected